fineract-client/src/main/java/org/apache/fineract/client/util/JSON.java

```diff
@@ -56,20 +57,64 @@ public class JSON {
     private final SqlDateTypeAdapter sqlDateTypeAdapter = new SqlDateTypeAdapter();
     private final OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter();
     private final LocalDateTypeAdapter localDateTypeAdapter = new LocalDateTypeAdapter();
+    private final LocalDateTimeTypeAdapter localDateTimeTypeAdapter = new LocalDateTimeTypeAdapter();
     private final ExternalIdAdapter externalIdAdapter = new ExternalIdAdapter();
 
     public JSON() {
         gson = new GsonFireBuilder().createGsonBuilder().registerTypeAdapter(Date.class, dateTypeAdapter)
                 .registerTypeAdapter(java.sql.Date.class, sqlDateTypeAdapter)
                 .registerTypeAdapter(OffsetDateTime.class, offsetDateTimeTypeAdapter)
-                .registerTypeAdapter(LocalDate.class, localDateTypeAdapter).registerTypeAdapter(ExternalId.class, externalIdAdapter)
+                .registerTypeAdapter(LocalDate.class, localDateTypeAdapter)
+                .registerTypeAdapter(LocalDateTime.class, localDateTimeTypeAdapter).registerTypeAdapter(ExternalId.class, externalIdAdapter)
                 .create();
     }
 
     public Gson getGson() {
         return gson;
     }
 
+    public static class LocalDateTimeTypeAdapter extends TypeAdapter<LocalDateTime> {
+
+        private final DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+
+        @Override
+        public void write(JsonWriter out, LocalDateTime date) throws IOException {
+            if (date == null) {
+                out.nullValue();
+            } else {
+                out.value(formatter.format(date));
+            }
+        }
+
+        @Override
+        public LocalDateTime read(JsonReader in) throws IOException {
+            switch (in.peek()) {
+                case NULL:
+                    in.nextNull();
+                    return null;
+                case STRING:
+                    String dateString = in.nextString();
+                    if (dateString != null && dateString.length() != 0) {
+                        return LocalDateTime.parse(dateString);
```

Comment:
Even though this is parsing in ISO_LOCAL_DATE_TIME format, since you already have a formatter in this class, I'd pass it to the parse method, just to make sure in case the format changes somehow, parsing doesn't break.

Reply:
LocalDateTime adapter was removed as suggested by @adamsaghy